_posts/2019-01-04-nextjs.md

```diff
@@ -0,0 +1,290 @@
+
+## Sample Project
+
+Let's create a simple project with Next.js to demonstrate how it works. You will be surprised how little we need to do in order to start our actual application. Some complexity is hidden by the library and the rule convention over configuration is helping to keep code on our side really slim.
+
+Create a new directory for our sample project `mkdir sample-app && cd sample-app`. Generate an empty npm project here with `npm init -y`. Then install Next.js together with React dependencies: `npm i next react react-dom --save`.
+
+Now we just have to add a new npm run script to our `package.json`:
+
+```js
+  "scripts": {
+    "dev": "next"
+  }
+```
+
+Let's try: `npm run dev`. Unfortunately you'll see an error message in your terminal. But with  next.js, error messages are usually pretty good and tell you exactly what you have to do next:
+
+`Couldn't find a `pages` directory. Please create one under the project root`
+
+So go ahead, create a `pages` directory and rerun `npm run dev`. It will tell you that it is now running on port 3000 - so open localhost:3000 in your browser. It's rendering a 404 page however because we did not yet create any "page".
+
+<img src="/img/posts/2019-01-04-nextjs/404.png" alt="404 Not Found" class="centered" />
+
+Lets create an `index.js` file with the following content:
+
+`export default () => <div>Hello World</div>`
+
+If you refresh your browser you'll see our new awesome "Hello World" page. This sounds like very little but what actually happened under the hood could take you quite some time if you really do it from scratch:
+
+- we got a free routing layer that is based on the convention that every path maps to a file in the pages folder (`/` is index.js, `/hello` is `hello.js`)
+- our code is automatically transpiled - we can use `export` without any hassle
+- we automatically can use react and `jsx` without any webpack setup
+- if you change the text from `Hello World` to `Hello World!` it will automatically be refreshed in your browser
+- ...
+
+
+But lets focus on the "application" we want to build.
+
+## Showing hotel data
+
+We want to show data about an hotel. So let's create the necessary files for that:
+
+For simplicities sake we will just use a static data file for the hotel data - src/hotel.js:
+```
+module.exports = {
+    id: '1aa4c4ad-f9ea-3367-a163-8a3a6884d450',
+    name: 'Dana Beach Resort',
+    description: 'Das modern ausgestattete 5-Sterne-Hotel Dana Beach Resort liegt direkt am eigenen Sandstrand, erbaut in einem einzigartigen Stil, mit hohem Komfort, mit bestem Service und einem umfangreichen Sport- & Freizeitangebot.'
```

Comment:
I would not go with German description ;) [Lorem ipsum](https://www.lipsum.com/) would also do fine here.